> > breakpad build failed on centos7 environment.
> > ```
> > 2025-06-17T17:50:39.4705575Z #9 2755.4 src/common/linux/dump_symbols.cc: In function ‘bool {anonymous}::IsCompressedHeader(const typename ElfClass::Shdr*)’:
> > 2025-06-17T17:50:39.4706359Z #9 2755.4 src/common/linux/dump_symbols.cc:298:31: error: ‘SHF_COMPRESSED’ was not declared in this scope
> > 2025-06-17T17:50:39.4706818Z #9 2755.4   298 |   return (section->sh_flags & SHF_COMPRESSED) != 0;
> > 2025-06-17T17:50:39.4707140Z #9 2755.4       |                               ^~~~~~~~~~~~~~
> > 2025-06-17T17:50:39.4707942Z #9 2755.4 src/common/linux/dump_symbols.cc: In function ‘std::pair<unsigned char*, long unsigned int> {anonymous}::UncompressSectionContents(uint64_t, const uint8_t*, uint64_t, uint64_t)’:
> > 2025-06-17T17:50:39.4708913Z #9 2755.4 src/common/linux/dump_symbols.cc:363:27: error: ‘ELFCOMPRESS_ZLIB’ was not declared in this scope; did you mean ‘ELFCOMPRESS_ZSTD’?
> > 2025-06-17T17:50:39.4709569Z #9 2755.4   363 |   if (compression_type == ELFCOMPRESS_ZLIB) {
> > 2025-06-17T17:50:39.4709871Z #9 2755.4       |                           ^~~~~~~~~~~~~~~~
> > 2025-06-17T17:50:39.4710155Z #9 2755.4       |                           ELFCOMPRESS_ZSTD
> > 2025-06-17T17:50:39.4710936Z #9 2755.4 src/common/linux/dump_symbols.cc: In function ‘bool {anonymous}::DwarfCFIRegisterNames(const typename ElfClass::Ehdr*, std::vector<std::__cxx11::basic_string<char> >*)’:
> > 2025-06-17T17:50:39.4711909Z #9 2755.4 src/common/linux/dump_symbols.cc:549:10: error: ‘EM_RISCV’ was not declared in this scope
> > 2025-06-17T17:50:39.4712295Z #9 2755.4   549 |     case EM_RISCV:
> > 2025-06-17T17:50:39.4712516Z #9 2755.4       |          ^~~~~~~~
> > 2025-06-17T17:50:39.4713097Z #9 2755.4 src/common/linux/dump_symbols.cc: In function ‘const char* {anonymous}::ElfArchitecture(const typename ElfClass::Ehdr*)’:
> > 2025-06-17T17:50:39.4713787Z #9 2755.4 src/common/linux/dump_symbols.cc:1125:10: error: ‘EM_RISCV’ was not declared in this scope
> > 2025-06-17T17:50:39.4714215Z #9 2755.4  1125 |     case EM_RISCV:      return "riscv";
> > 2025-06-17T17:50:39.4714473Z #9 2755.4       |          ^~~~~~~~
> > ```
> 
> @kevincai these macros are present in /usr/include/elf.h. unavailable on centos7 since its pretty old. dont observe the same failure on ubuntu. how about a patch file to add these macros to dump_symbols.cc?

Yes, it's acceptable.

--- a/thirdparty/build-thirdparty.sh
+++ b/thirdparty/build-thirdparty.sh
@@ -915,7 +915,7 @@ build_breakpad() {
     cd $TP_SOURCE_DIR/$BREAK_PAD_SOURCE
     mkdir -p src/third_party/lss
     cp $TP_PATCH_DIR/linux_syscall_support.h src/third_party/lss
-    LDFLAGS="-L${TP_INSTALL_DIR}/lib -L${TP_INSTALL_DIR}/lib64" \
+    LDFLAGS="-L${TP_LIB_DIR}" \
     CFLAGS= ./configure --prefix=$TP_INSTALL_DIR --enable-shared=no --disable-samples --disable-libevent-regress
     make -j$PARALLEL
     make install

--- a/thirdparty/build-thirdparty.sh
+++ b/thirdparty/build-thirdparty.sh
@@ -915,6 +915,7 @@ build_breakpad() {
     cd $TP_SOURCE_DIR/$BREAK_PAD_SOURCE
     mkdir -p src/third_party/lss
     cp $TP_PATCH_DIR/linux_syscall_support.h src/third_party/lss
+    LDFLAGS="-L${TP_INSTALL_DIR}/lib -L${TP_INSTALL_DIR}/lib64" \
     CFLAGS= ./configure --prefix=$TP_INSTALL_DIR --enable-shared=no --disable-samples --disable-libevent-regress
     make -j$PARALLEL
     make install

--- a/thirdparty/download-thirdparty.sh
+++ b/thirdparty/download-thirdparty.sh
@@ -596,6 +596,10 @@ if [[ -d $TP_SOURCE_DIR/$BREAK_PAD_SOURCE ]] ; then
         patch -p1 < "$TP_PATCH_DIR/breakpad-2022.07.12.patch"
         touch "$PATCHED_MARK"
     fi
+    if [ ! -f "$PATCHED_MARK" ] && [[ $BREAK_PAD_SOURCE == "breakpad-2024.02.16" ]] ; then
+        patch -p1 < "$TP_PATCH_DIR/breakpad-2024.02.16.patch"
+        touch "$PATCHED_MARK"
+    fi
     cd -
     echo "Finished patching $BREAK_PAD_SOURCE"
 fi

--- a/thirdparty/minidump/gen_minidump_symbols.sh
+++ b/thirdparty/minidump/gen_minidump_symbols.sh
@@ -12,10 +12,10 @@ starrocks_home=$(dirname $(dirname $curdir))
 # generate symbol's file in Breakpad's own format.
 $(dirname $curdir)/installed/bin/dump_syms $starrocks_home/output/be/lib/starrocks_be > $starrocks_home/output/be/starrocks_be.sym
 
-((starrocks_be_size_original=`ls -l output/be/lib/starrocks_be | awk '{print $5}'` / (1024*1024)))
+((starrocks_be_size_original=`ls -l $starrocks_home/output/be/lib/starrocks_be | awk '{print $5}'` / (1024*1024)))
 # remove debugging infos
 strip $starrocks_home/output/be/lib/starrocks_be
-((starrocks_be_size_simplify=`ls -l output/be/lib/starrocks_be | awk '{print $5}'` / (1024*1024)))
+((starrocks_be_size_simplify=`ls -l $starrocks_home/output/be/lib/starrocks_be | awk '{print $5}'` / (1024*1024)))
 
 # echo size reduction
 echo "starrocks_be'size ("$starrocks_be_size_original"mb) reduced to ("$starrocks_be_size_simplify"mb)"

--- a/thirdparty/patches/breakpad-2024.02.16.patch
+++ b/thirdparty/patches/breakpad-2024.02.16.patch
@@ -1,0 +1,17 @@
+diff --git a/src/common/linux/dump_symbols.cc b/src/common/linux/dump_symbols.cc
+index b693fc9e..8686ee38 100644
+--- a/src/common/linux/dump_symbols.cc
++++ b/src/common/linux/dump_symbols.cc
+@@ -82,6 +82,12 @@
+ #endif
+ #include "common/using_std_string.h"
+ 
++#ifndef SHF_COMPRESSED
++#define SHF_COMPRESSED       (1 << 11)  /* Section with compressed data. */
++#define ELFCOMPRESS_ZLIB     1          /* ZLIB/DEFLATE algorithm.  */
++#define EM_RISCV             243     /* RISC-V */
++#endif
++
+ // This namespace contains helper functions.
+ namespace {
+ 

--- a/thirdparty/vars.sh
+++ b/thirdparty/vars.sh
@@ -295,10 +295,10 @@ RYU_SOURCE="ryu-aa31ca9361d21b1a00ee054aac49c87d07e74abc"
 RYU_MD5SUM="cb82b6da904d919470fe3f5a01ca30ff"
 
 # breakpad
-BREAK_PAD_DOWNLOAD="https://github.com/google/breakpad/archive/refs/tags/v2022.07.12.tar.gz"
-BREAK_PAD_NAME="breakpad-2022.07.12.tar.gz"
-BREAK_PAD_SOURCE="breakpad-2022.07.12"
-BREAK_PAD_MD5SUM="d5bcfd3f7b361ef5bda96123c3abdd0a"
+BREAK_PAD_DOWNLOAD="https://github.com/google/breakpad/archive/refs/tags/v2024.02.16.tar.gz"
+BREAK_PAD_NAME="breakpad-2024.02.16.tar.gz"
+BREAK_PAD_SOURCE="breakpad-2024.02.16"
+BREAK_PAD_MD5SUM="ae8c55b23c157771922b5ddca3803055"
 
 # RAGEL
 # ragel-6.9+ is used by hyperscan, so we build it first

